adflow/pyADflow.py

```diff
@@ -3991,19 +3994,26 @@ def computeJacobianVectorProductFwd(
         costSize = self.adflow.constants.ncostfunction
         fSize, nCell = self._getSurfaceSize(self.allWallsGroup, includeZipper=True)
 
+        # process the functions and groups
+        if evalFuncs is None:
+            evalFuncs = sorted(self.curAP.evalFuncs)
+
         # Generate the list of families we need for the functions in curAP
-        groupNames = set()
-        for f in self.curAP.evalFuncs:
+        # We need to use a list to have the same ordering on every proc, but
+        # we need 'set like' behavior so we don't include duplicate group names.
+        groupNames = []
```

Comment:
This makes sense, it looks like the bwd version was already using a similar approach